src/ripple/server/impl/Door.h

```diff
@@ -221,32 +224,30 @@ Door<Handler>::Detector::do_detect(boost::asio::yield_context do_yield)
 //------------------------------------------------------------------------------
 
 template <class Handler>
-Door<Handler>::Door(
-    Handler& handler,
-    boost::asio::io_context& io_context,
-    Port const& port,
-    beast::Journal j)
-    : j_(j)
-    , port_(port)
-    , handler_(handler)
-    , ioc_(io_context)
-    , acceptor_(io_context)
-    , strand_(io_context)
-    , ssl_(
-          port_.protocol.count("https") > 0 ||
-          port_.protocol.count("wss") > 0 || port_.protocol.count("wss2") > 0 ||
-          port_.protocol.count("peer") > 0)
-    , plain_(
-          port_.protocol.count("http") > 0 || port_.protocol.count("ws") > 0 ||
-          port_.protocol.count("ws2"))
+void
+Door<Handler>::reOpen()
```

Comment:
Why `reOpen`? And why make this a function?

Reply:
@mtrippled can you comment on this?

Reply:
@nbougalis @cjcobb23 During testing I found that the server entered into an endless loop during accept with the error of too many open files, even after the sockets were disconnected, no longer existing. Basically, based on the location of the error message in the code, it was in the acceptor logic. Basically, the socket spawned by accept() was emitting that error in an endless loop, even well after the filehandles were freed based on netstat -an output. So I created the function to idempotently close then reopen the acceptor in the event of this thing happening again. I didn't reproduce this. The behavior, to me, seemed like a bug in the OS somewhere. accept() kept trying to open a new socket even with no incoming connections. It's possible there was a listen() backlog, but the backlog is generally 4096 unless overridden (see listen(2)). The infinite loop of "too many open files" errors emitted was easily in the 10's of thousands, probably millions by the time I detected it.

So, the reason for this function is to hopefully avoid an endless loop in a very specific high traffic situation that's hard to reproduce--and it might not even fix the issue since I didn't reproduce this.